### DISCUSSION TITLE
Vendor lock and other cloud questions

### DISCUSSION THREAD

**Discussion by narcis-ro** [zitadel/zitadel#5516] [General]
Hello,

First of all congratz on what looks to be an awesome product.

We’re evaluating if we should use Zitadel for a rewrite of our saas back-office portal. We’re evaluating the features as well as the cloud solution. Zitadel fits perfect for a b2b/b2c mix.

We’ll probably talk to the sales team at some point but in the meantime if you coud use share some light in some of our concerns:

- Using the cloud solution, is it possible/easy to migrate to a self-hosted option? Our worries are that pricing might get unsustainable high in the future.
- Any plans to officially support Azure cloud provider and/or offer managed services in Azure? Being able to choose a closer datacenter zone would be nice in order to keep latency low. Ability to use vnet peering would be nice.
- The number of action minutes in the cloud ofering seems a bit low. If it were to make a call to a external api, with some retries we could easily exceed 60 minutes a month and sure you could buy more minutes, but that can get out of control quickly. I also wonder if there is cap/warning to not bankrupt us :)
- Any idea when Postgres would come out of beta? Azure has a managed Postgres offering and if we were to go with the self hosted version we would prefer not to manage other infra ourselves.
- We would need the interface for login/self service (register, forgot etc) to be translated in romanian and we are wiling to do that. Do we need to translate the whole console as well or can we submit PR for just client facing views? Are there any guidelines to PR translation? Or just follow previous PRs for other languages
- For different environments, would you recommend to has one account for development with sub environementd (dev, test, staging) maybe as free tier and another one for production (preprod and prod). Or could we model this with one cloud account without any risks.
- Authentik has this nice stage for captcha. Do you have/plan to support this feature, at least for registrion

Sorry for the long post!

P.S As a feedback to the product team. Keep doing an awesome job with the documentation, samples and examples, from code, config to use cases. These mean the world to everyone trying to pick one provider from a red sea of authN providers. This is were auth0 shines, but their hidden/confusing pricing suck.

**Comment by mffap** [comment]
> Using the cloud solution, is it possible/easy to migrate to a self-hosted option? Our worries are that pricing might get unsustainable high in the future.

Two parts to that answer: Yes it is possible to move between the cloud and self-hosted. It's the identical software and the data can be imported/exported (eg, with terraform). Typically for high-usage clients we do offer a pricing that better suits their needs. This can be for example a tiered pricing along a baseline / low load / high load / spikes scenario.

> Any plans to officially support Azure cloud provider and/or offer managed services in Azure? Being able to choose a closer datacenter zone would be nice in order to keep latency low. Ability to use vnet peering would be nice.

No we don't offer a multi-provider strategy. We monitor latency on our 100+ entry notes and optimize latency according to traffic and other factors. If you want to host on Azure you can do so (we have customers running on Azure) and get support from our engineers with the enterprise license.

> The number of action minutes in the cloud ofering seems a bit low. If it were to make a call to a external api, with some retries we could easily exceed 60 minutes a month and sure you could buy more minutes, but that can get out of control quickly. I also wonder if there is cap/warning to not bankrupt us :)

It's 60 minutes per billing unit. We haven't seen that threshold to be exceeded with "normal" use. If you're concerned about the limits, please contact us at hi@zitadel.com with your customer details (instance id).

> Any idea when Postgres would come out of beta? Azure has a managed Postgres offering and if we were to go with the self hosted version we would prefer not to manage other infra ourselves.

Did you encounter any issues when trying postgres? Our [recommended DB](https://zitadel.com/docs/self-hosting/manage/production#prefer-cockroachdb) will remain CockroachDB. However we see our community using postgres. We will clarify how we move out of the beta on our docs (@adlerhurst, @hifabienne) :)

> We would need the interface for login/self service (register, forgot etc) to be translated in romanian and we are wiling to do that. Do we need to translate the whole console as well or can we submit PR for just client facing views? Are there any guidelines to PR translation? Or just follow previous PRs for other languages

We give some hints in our [contribution guide](https://github.com/zitadel/zitadel/blob/main/CONTRIBUTING.md#contribute-internationalization), but I guess easiest would be to follow some of the PRs, like the recent [Japanese](https://github.com/zitadel/zitadel/issues/4233) translation.
Unfortunately the translations must be done for all parts. We have a [task on our backlog](https://github.com/zitadel/zitadel/pull/5513)to split this, and I've added a link to this discussion for future prioritization.

> For different environments, would you recommend to has one account for development with sub environementd (dev, test, staging) maybe as free tier and another one for production (preprod and prod). Or could we model this with one cloud account without any risks.

Typically we see [users having different instances](https://github.com/zitadel/zitadel/issues/5512) for pre-prod environments. You could also use our e2e harness with a self-hosted dev environment to do automated testing before moving to stage...

> Authentik has this nice stage for captcha. Do you have/plan to support this feature, at least for registrion

Do you mean to on the login or on registering users? I assume the [stage for captcha](https://goauthentik.io/docs/flow/stages/captcha/) is during authentication. We use a strong WAF on our cloud service that would anyways trigger reCAPTCHA Enterprise to mitigate bots or prevent DDoS attacks.

> Sorry for the long post!

No worries - good questions :)

> P.S As a feedback to the product team. Keep doing an awesome job with the documentation, samples and examples, from code, config to use cases. These mean the world to everyone trying to pick one provider from a red sea of authN providers. This is were auth0 shines, but their hidden/confusing pricing suck.

Thanks so much for the feedback! We're working on more Examples and SDKs in different languages. Hope they land soon.
